rp2040-hal/src/rom_data.rs

```diff
@@ -0,0 +1,379 @@
+//! Functions and data from the RPI Bootrom.
+
+/// A bootrom function table code.
+pub type RomFnTableCode = [u8; 2];
+
+/// This function searches for (table)
+type RomTableLookupFn<T> = unsafe extern "C" fn(*const u16, u32) -> T;
+
+/// The following addresses are described at `2.8.3. Bootrom Contents`
+/// Pointer to the lookup table function supplied by the rom.
+const ROM_TABLE_LOOKUP_PTR: *const u16 = 0x18 as _;
```

Comment:
couldn't you just type this const as a function instead of doing a transmute?

Reply:
A function that returns `T`?

Reply:
oh right its generic. nvm.